projects/RabbitMQ.Client/client/FrameworkExtension/Interlocked.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace RabbitMQ.Client
+{
+#if NETCOREAPP3_1 || NETSTANDARD
+    internal static class Interlocked
```

Comment:
These methods do not exist in < ".NET 5", to easy future migration, I've created this class and copied the implementation of the framework.

Reply:
Would this require a license header that complies with MS licensing?

Reply:
Good question, I'm not the expert here for this. Who can check this?

Reply:
The .NET reference browser doesn't show these exact methods in place, and neither does the code in `dotnet/runtime`. There is very similar code in the latter in this file:

```
src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Interlocked.cs
```

The header of the file reads:

```
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
```

So, the code that @bollhals wrote could be considered to be inspired by the latter. I will make a note of it.